### PR TITLE
Adding `gradsflow` from PyPI

### PR DESCRIPTION
**Pull request by sugatoray**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged. `NA`
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by sugatoray** [comment]
Missing requirements threw [error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=425458&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1149).

```sh
1149 ModuleNotFoundError: No module named 'flit_core'
```

Adding `flit-core` to requirements in recipe: https://anaconda.org/conda-forge/flit-core.

**Comment by sugatoray** [comment]
:fire: Missing requirements threw [error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=425464&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1521).

```sh
1521 ModuleNotFoundError: No module named 'pytorch_lightning'
```

- Adding `comet` to requirements in recipe: anaconda.org/conda-forge/pytorch-lightning

---

:fire: Missing requirements threw [error](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=425464&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1521).

```sh
1584 ModuleNotFoundError: No module named 'comet_ml'
```

- Adding `comet` to requirements in recipe: https://anaconda.org/conda-forge/comet. 
- This conda-forge package was last updated two years ago (v.3.0.0). A more current version is available on `comet_ml` channel: https://anaconda.org/comet_ml/comet_ml.

**Comment by sugatoray** [comment]
## :bulb: Reasons for CI failing

- Build failure on **osx**: `ray-tune` and `ray-default` are not available for `osx` on `conda-forge` (source:[ray-tune][#cf-ray-tune], [ray-default][#cf-ray-default])

- Build failure on **win**: `pytorch` is not available for `win` on `conda-forge` (source: [pytorch][#cf-pytorch])

[#cf-pytorch]: https://anaconda.org/conda-forge/pytorch
[#cf-ray-tune]: https://anaconda.org/conda-forge/ray-tune
[#cf-ray-default]: https://anaconda.org/conda-forge/ray-default

<div>
<details>
<summary>OS Availability: <code>pytorch</code></summary>
<p>

| ![pytorch][#cf-pytorch-img] | 
|:---:|
| pytorch |

<p>
</details>

<details>
<summary>OS Availability: <code>ray-tune</code></summary>
<p>

| ![ray-tune][#cf-ray-tune-img] |
|:---:|
| ray-tune |

<p>
</details>

<details>
<summary>OS Availability: <code>ray-default</code></summary>
<p>

| ![ray-default][#cf-ray-default-img] |
|:---:|
| ray-default |

<p>
</details>
</div>

[#cf-pytorch-img]: https://user-images.githubusercontent.com/10201242/145707179-fc089121-7309-4032-989f-897f1aaba52e.png

[#cf-ray-tune-img]: https://user-images.githubusercontent.com/10201242/145707197-c6300e3a-aadf-4f71-9e3b-e51796c07c88.png

[#cf-ray-default-img]: https://user-images.githubusercontent.com/10201242/145707213-23f17be0-9ab7-48e1-bf82-299a62cb9d52.png

**Comment by sugatoray** [comment]
Ready for review.

cc: @conda-forge/help-python

File: recipes/gradsflow/meta.yaml
Comment (dopplershift) [inline recipes/gradsflow/meta.yaml:37]: With comet-ml being optional (and not available from conda-forge) this should be removed. (With no version it wasn't doing anything anyway.)
```suggestion
```

File: recipes/gradsflow/meta.yaml
Comment (dopplershift) [inline recipes/gradsflow/meta.yaml:41]: Go ahead and explicitly list the name here. If the package fully changes the import, odds are we want a new package anyway.
```suggestion
    - gradsflow
```

File: recipes/gradsflow/meta.yaml
Comment (dopplershift) [inline recipes/gradsflow/meta.yaml:44]: This needs to go
```suggestion
```

File: recipes/gradsflow/meta.yaml
Comment (dopplershift) [inline recipes/gradsflow/meta.yaml:26]: Should either remove completely or uncomment (and remove under test).

**Review by dopplershift — CHANGES_REQUESTED** [review]

File: recipes/gradsflow/meta.yaml
Comment (sugatoray) [reply recipes/gradsflow/meta.yaml]: Done.

File: recipes/gradsflow/meta.yaml
Comment (sugatoray) [reply recipes/gradsflow/meta.yaml]: Done.

File: recipes/gradsflow/meta.yaml
Comment (sugatoray) [reply recipes/gradsflow/meta.yaml]: Done.

**Comment by sugatoray** [comment]
## Remove `tests/` folder from package

The PyPI version of the package includes `tests/` folder. The library maintainer(s) have been informed. Issue was raised to remove the `tests/` folder from `gradsflow`.

Issue: https://github.com/gradsflow/gradsflow/issues/140

image

File: recipes/gradsflow/meta.yaml
Comment (sugatoray) [reply recipes/gradsflow/meta.yaml]: Thank you. Done.

**Comment by sugatoray** [comment]
@dopplershift Thank you, for the thorough review. I have incorporated all the suggestions you made. Please let me know if there is anything else that needs changing before this could be merged.

**Review by dopplershift — APPROVED** [review]
Ignoring noarch win/osx failures due to missing dependencies.

**Pull request merged**
